rust/hyperlane-base/src/agent/metadata.rs

```diff
@@ -0,0 +1,16 @@
+use serde::{Deserialize, Serialize};
+
+/// Metadata about agent
+#[derive(Debug, Deserialize, Serialize)]
+pub struct AgentMetadata {
```

Comment:
nit: we use the `derive_new` macro throughout the codebase to generate constructor boilerplate, you could derive it by adding `new` here imported from `derive_new::new`. Then you could remove the `impl AgentMetadata`

Reply:
Nice. Good catch!